flake.nix

```diff
@@ -24,211 +25,299 @@
-          inherit (project.args) src;
-        };
+          # Set up and start Postgres before running database tests
+          preCheck = ''
+            echo pre-check
+            initdb --encoding=UTF8 --locale=en_US.UTF-8 --username=postgres $NIX_BUILD_TOP/db-dir
+            postgres -D $NIX_BUILD_TOP/db-dir -c listen_addresses="" -k $TMP &
+            PSQL_PID=$!
+            echo $PSQL_PID > postgres.pid
+            sleep 10
+            if (echo '\q' | psql -h $TMP postgres postgres); then
+              echo "PostgreSQL server is verified to be started."
+            else
+              echo "Failed to connect to local PostgreSQL server."
+              exit 2
+            fi
+            ls -ltrh $NIX_BUILD_TOP
+            DBUSER=$(whoami)
+            DBNAME=$DBUSER
+            export PGPASSFILE=$NIX_BUILD_TOP/pgpass
+            echo "$TMP:5432:$DBUSER:$DBUSER:*" > $PGPASSFILE
+            cp -vir ${./schema} ../schema
+            chmod 600 $PGPASSFILE
+            psql -h $TMP postgres postgres <<EOF
+              create role $DBUSER with createdb login password '$DBPASS';
+              alter user $DBUSER with superuser;
+              create database $DBNAME with owner = $DBUSER;
+              \\connect $DBNAME
+              ALTER SCHEMA public   OWNER TO $DBUSER;
+            EOF
+          '';
 
-        nixosTests = import ./nix/nixos/tests {
-          inherit pkgs;
-        };
+          # Clean up Postgres after running database tests
+          postCheck = ''
+            echo post-check
+            DBNAME=$(whoami)
+            NAME=db_schema.sql
+            mkdir -p $out/nix-support
+            echo "Dumping schema to db_schema.sql"
+            pg_dump -h $TMP -s $DBNAME > $out/$NAME
+            kill $(cat postgres.pid)
+            echo "Adding to build products..."
+            echo "file binary-dist $out/$NAME" > $out/nix-support/hydra-build-products
+          '';
 
-        checks = project.flake'.checks // {
-          inherit hlint fourmolu shellcheck;
-        } // lib.optionalAttrs hostPlatform.isLinux (prefixNamesWith "nixosTests/" nixosTests);
+          project = (nixpkgs.haskell-nix.cabalProject' ({ config, lib, ... }: rec {
+            src = ./.;
+            name = "cardano-db-sync";
+            compiler-nix-name = lib.mkDefault "ghc8107";
+            flake.variants =
+              lib.genAttrs
+                ["ghc962"]
+                (compiler-nix-name: { inherit compiler-nix-name; });
 
-        # This is used by `nix develop` to open a devShell
-        devShells.default = project.shell;
+            inputMap = {
+              "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
+            };
 
-        scripts = flattenTree pkgs.scripts;
+            shell.buildInputs = with nixpkgs.pkgsBuildBuild; [
+              haskell-language-server
+              gitAndTools.git
+            ];
+            shell.withHoogle = true;
```

Comment:
this is a different shell from the devx shell, do we need to add devx as a flake dependency?